kotlinx-coroutines-core/nativeDarwin/src/Dispatchers.kt

```diff
@@ -28,7 +28,7 @@ private class DarwinMainDispatcher(
 
     init { freeze() }
 
-    override fun isDispatchNeeded(context: CoroutineContext): Boolean = !invokeImmediately || isMainThread()
+    override fun isDispatchNeeded(context: CoroutineContext): Boolean = !(invokeImmediately && isMainThread())
```

Comment:
The function call style `.not()` operator would probably be more readable than `!` here.

Reply:
I don't think so. It might more readable if you distribute negation, though: `!invokeImmediately || !isMainThread()`